src/mono/browser/runtime/marshal-to-cs.ts

```diff
@@ -261,13 +262,14 @@ function _marshal_function_to_cs(arg: JSMarshalerArgument, value: Function, _?:
     mono_check(value && value instanceof Function, "Value is not a Function");
 
     // TODO: we could try to cache value -> existing JSHandle
-    const wrapper: any = (args: JSMarshalerArguments) => {
+    const wrapper: any = function delegate_wrapper(args: JSMarshalerArguments) {
         const exc = get_arg(args, 0);
         const res = get_arg(args, 1);
         const arg1 = get_arg(args, 2);
         const arg2 = get_arg(args, 3);
         const arg3 = get_arg(args, 4);
```

Comment:
`ActionJS.InvokeJS` stackallocs JSMarshalerArgument[4], but here we're calling `get_arg` on indices `0-4`, which would require a buffer of size 5. I can't find anywhere we allocate a buffer that big, and `get_arg` doesn't do any bounds checks. Is it right for all of this to be written this way or are we potentially dereferencing random memory?

Reply:
The C# signatures know the number of parameters and so the allocation should be different. But you are onto something, I just made couple of them smaller. Like 2 for `fn (void) =>void`